src/olympia/stats/utils.py

```diff
@@ -0,0 +1,69 @@
+from django.conf import settings
+from google.cloud import bigquery
+
+from olympia.constants.applications import FIREFOX
+
+# This is the mapping between the AMO stats `sources` and the BigQuery columns.
+AMO_TO_BIGQUERY_COLUMN_MAPPING = {
+    'apps': 'dau_by_app_version',
+    'locales': 'dau_by_locale',
+    'os': 'dau_by_app_os',
+    'versions': 'dau_by_addon_version',
+}
+
+
+def rows_to_series(rows, filter_by=None):
+    """Transforms BigQuery rows into series items suitable for the rest of the
+    AMO stats logic."""
+    for row in rows:
+        item = {
+            'count': row.dau,
+            'date': row.submission_date,
+            'end': row.submission_date,
+        }
+        if filter_by:
+            item['data'] = {d['key']: d['value'] for d in row[filter_by]}
+
+            # See: https://github.com/mozilla/addons-server/issues/14411
+            if filter_by == 'dau_by_app_version':
+                item['data'] = {FIREFOX.guid: item['data']}
+
+        yield item
+
+
+def get_updates_series(addon, start_date, end_date, source=None):
+    client = bigquery.Client.from_service_account_json(
+        settings.GOOGLE_APPLICATION_CREDENTIALS
+    )
+
+    filter_by = AMO_TO_BIGQUERY_COLUMN_MAPPING[source] if source else None
+
+    select_clause = 'SELECT submission_date, dau'
+    if filter_by:
+        select_clause = f'{select_clause}, {filter_by}'
+    query = f"""
+{select_clause}
+FROM `moz-fx-data-shared-prod.telemetry.amo_stats_dau`
```

Comment:
This table name might change but it's not clear how or when. This is the production table, though so that will work in prod.